files/image_config/caclmgrd/caclmgrd

```diff
@@ -247,6 +304,63 @@ class ControlPlaneAclManager(object):
+                    iptables_cmds.append("ip6tables -A INPUT -p {0} --dport {1} -j ACCEPT".format(ip_protocol, dst_port))
+
+        # Add iptables rules to drop all packets destined for loopback interface IP addresses
+        # (and in the case of V6, all point-to-point addresses in the /64 subnet)
+        loopback_iface_table = self.config_db.get_table(LOOPBACK_INTERFACE_TABLE)
+        for ((iface_name, iface_cidr), _) in loopback_iface_table.iteritems():
+            ip_ntwrk = ipaddress.ip_network(iface_cidr, strict=False)
+            if isinstance(ip_ntwrk, ipaddress.IPv4Network):
+                iptables_cmds.append("iptables -A INPUT -d {}/32 -j DROP".format(ip_ntwrk.network_address))
+            elif isinstance(ip_ntwrk, ipaddress.IPv6Network):
+                iptables_cmds.append("ip6tables -A INPUT -d {}/64 -j DROP".format(ip_ntwrk.network_address))
+            else:
+                log_warning("Unrecognized IP address type: {}".format(ip_ntwrk))
+
+        # Add iptables rules to drop all packets destined for management interface IP addresses
+        mgmt_iface_table = self.config_db.get_table(MGMT_INTERFACE_TABLE)
+        for ((iface_name, iface_cidr), _) in mgmt_iface_table.iteritems():
+            ip_ntwrk = ipaddress.ip_network(iface_cidr, strict=False)
+            if isinstance(ip_ntwrk, ipaddress.IPv4Network):
+                iptables_cmds.append("iptables -A INPUT -d {}/32 -j DROP".format(ip_ntwrk.network_address))
+            elif isinstance(ip_ntwrk, ipaddress.IPv6Network):
+                iptables_cmds.append("ip6tables -A INPUT -d {}/128 -j DROP".format(ip_ntwrk.network_address))
+            else:
+                log_warning("Unrecognized IP address type: {}".format(ip_ntwrk))
+
+        # Add iptables rules to drop all packets destined for our VLAN interface gateway IP addresses
+        vlan_iface_table = self.config_db.get_table(VLAN_INTERFACE_TABLE)
+        for ((iface_name, iface_cidr), _) in vlan_iface_table.iteritems():
+            ip_ntwrk = ipaddress.ip_network(iface_cidr, strict=False)
+            if isinstance(ip_ntwrk, ipaddress.IPv4Network):
+                iptables_cmds.append("iptables -A INPUT -d {}/32 -j DROP".format(list(ip_ntwrk.hosts())[0]))
+            elif isinstance(ip_ntwrk, ipaddress.IPv6Network):
+                iptables_cmds.append("ip6tables -A INPUT -d {}/128 -j DROP".format(list(ip_ntwrk.hosts())[0]))
+            else:
+                log_warning("Unrecognized IP address type: {}".format(ip_ntwrk))
+
+        # Add iptables/ip6tables commands to allow all incoming packets with TTL of 0 or 1
+        # This allows the device to respond to tools like tcptraceroute
+        iptables_cmds.append("iptables -A INPUT -m ttl --ttl-lt 2 -j ACCEPT")
+        iptables_cmds.append("ip6tables -A INPUT -p tcp -m hl --hl-lt 2 -j ACCEPT")
+
+        # Finally, add iptables/ip6tables commands to drop all other incoming packets
```

Comment:
in case none of the acl service is configured, should we drop this default deny rule at all?

Reply:
I considered this approach, but instead of allowing all packets, I added the loop starting on line 310 to allow all incoming traffic for each unconfigured service.

Reply:
Changed in latest commit. Now we apply the default deny all rules only if we found 1 or more configured control plane ACL rules.